components/arm/wrapper/wrapper.go

```diff
@@ -176,13 +176,19 @@ func (wrapper *Arm) CurrentInputs(ctx context.Context) ([]referenceframe.Input,
 }
 
 // GoToInputs moves the arm to the specified goal inputs.
-func (wrapper *Arm) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// check that joint positions are not out of bounds
-	positionDegs := wrapper.model.ProtobufFromInput(goal)
-	if err := arm.CheckDesiredJointPositions(ctx, wrapper, positionDegs); err != nil {
-		return err
+func (wrapper *Arm) GoToInputs(ctx context.Context, inputSteps ...[]referenceframe.Input) error {
+	for _, goal := range inputSteps {
+		// check that joint positions are not out of bounds
+		positionDegs := wrapper.model.ProtobufFromInput(goal)
+		if err := arm.CheckDesiredJointPositions(ctx, wrapper, positionDegs); err != nil {
```

Comment:
Noticing now that this function takes in joint positions only to convert them back to limits to do the check.  Can you change this function as a part of this PR to take inputs instead?  Could even move it out of the arm package to the frame package and make it operate on the new `Limited` interface

Reply:
`arm.CheckDesiredJointPositions` is a little special; it's not *just* checking whether the given inputs are OOB, it has some arm-specific behavior that we probably don't want to generalize.

It will query an arm's current joint positions, check if *those* are OOB, and if so, if the requested joint positions move the OOB joints closer to in bounds, it permits that (even if it is still OOB).

I can update to take inputs, but recommend against moving.